### ISSUE TITLE
[MNG-5830] maven.multiModuleProjectDirectory system propery is not set

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7794]
**[Krzysztof Sobkowiak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sobkowiak)** opened **[MNG-5830](https://issues.apache.org/jira/browse/MNG-5830?redirect=false)** and commented

I have downloaded the newest Maven 3.3.3 and point the M2_HOME to the directory with unpacked Maven.  `mvn -v` produces following error (whereas the versions prior to 3.1.1 worked correctly)

```
kso@lenovo:/tmp$ export M2_HOME=/home/kso/work/pde/dev/maven/apache-maven-3.3.3/
kso@lenovo:/tmp$ mvn -v
-Dmaven.multiModuleProjectDirectory system propery is not set. Check $M2_HOME environment variable and mvn script match.kso@lenovo:/tmp$ 

```

The same problem when I want to build any maven project.

Below the same output with Maven 3.2.5 to give you informations about my environment

```
kso@lenovo:/tmp$ export M2_HOME=/home/kso/work/pde/dev/maven/apache-maven-3.2.5/
kso@lenovo:/tmp$ mvn -v
Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T18:29:23+01:00)
Maven home: /home/kso/work/pde/dev/maven/apache-maven-3.2.5
Java version: 1.7.0_71, vendor: Oracle Corporation
Java home: /home/kso/work/pde/run/jvm/jdk1.7.0_71/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.16.0-38-generic", arch: "amd64", family: "unix"

```

---

**Affects:** 3.3.1, 3.3.3

3 votes, 11 watchers

**Comment by jira-importer** [comment]
**[Matthew Boyd](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cizer)** commented

I was able to resolve this issue by adding the following to my .bash_profile:
export MAVEN_OPTS="-Dmaven.multiModuleProjectDirectory=$M2_HOME"

**Comment by jira-importer** [comment]
**[Krzysztof Sobkowiak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sobkowiak)** commented

Yes, I know this. But I don't thing the solution should look like this.

**Comment by jira-importer** [comment]
**[Ralph Goers](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ralph.goers@dslextreme.com)** commented

How did you determine that that was the correct value to supply?

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

What is in your path? The problem i assume is that you have an older version of maven in your PATH (PATH=..:/.../apache-maven-3.2.5/bin/ for example Maven 3.2.5) but you would like to use Maven 3.3.X where the shell script become some needed changes to support other things...

**Comment by jira-importer** [comment]
**[Ralph Goers](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ralph.goers@dslextreme.com)** commented

I'd be surprised if that was the case for buildbot.  Please see https://ci.apache.org/builders/log4j2-nightly/builds/98/steps/compile/logs/stdio

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Based on the error message i would assume so...cause the mvn script of mvn 3.3.X needs the variable multiModuleProjectDirectory which is introduced into Maven 3.3.X....so there is a mismatch between those things...BTW: is `/home/buildslave2/slave2/tools/maven/latest3/` pointing to correct location...

**Comment by jira-importer** [comment]
**[Ralph Goers](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ralph.goers@dslextreme.com)** commented

I have no way of knowing. I don't believe I have access to the buildbot server.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

If i take a look into the log file i can see that maven is called but the path does not contains a reference to Maven so...Neither i do have access to the buildbot server...

**Comment by jira-importer** [comment]
**[Christian Schulte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schulte77)** commented

I just downloaded Maven 3.3.1 and 3.3.3 from [here](http://archive.apache.org/dist/maven/maven-3/).

```
$ env M2_HOME=/tmp/apache-maven-3.3.1 /tmp/apache-maven-3.3.1/bin/mvn -v   
Apache Maven 3.3.1 (cab6659f9874fa96462afef40fcf6bc033d58c1c; 2015-03-13T21:10:27+01:00)
Maven home: /tmp/apache-maven-3.3.1
Java version: 1.8.0_45, vendor: Oracle Corporation
Java home: /usr/local/jdk-1.8.0/jre
Default locale: en_DE, platform encoding: UTF-8
```

```
$ env M2_HOME=/tmp/apache-maven-3.3.3 /tmp/apache-maven-3.3.3/bin/mvn -v
Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T13:57:37+02:00)
Maven home: /tmp/apache-maven-3.3.3
Java version: 1.8.0_45, vendor: Oracle Corporation
Java home: /usr/local/jdk-1.8.0/jre
Default locale: en_DE, platform encoding: UTF-8
```

Not reproducible.

**Comment by jira-importer** [comment]
**[Markus Karg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mkarg)** commented

The Eclipse Foundation has this problem (maven.multiModuleProjectDirectory not getting resolved) with Maven 3.5.4! [Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise) can you please help us fixing this on our public EF build infrastructure? This is a showstopper for Jakarta EE 8. Would be great, thanks! :)

**Comment by jira-importer** [comment]
**[Martynas Jusevičius](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=graphity)** commented

Happens to me after I tried upgrading Maven to 3.6.2 using Chocolatey on cmder (Windows 10): https://chocolatey.org/packages/maven#upgrade
